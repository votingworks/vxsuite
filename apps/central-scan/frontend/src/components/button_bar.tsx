import styled from 'styled-components';

interface Props {
  dark?: boolean;
  naturalOrder?: boolean;
  secondary?: boolean;
  separatePrimaryButton?: boolean;
  centerOnlyChild?: boolean;
}

export const ButtonBar = styled('nav')<Props>`
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  order: ${({ secondary }) => (secondary ? '-1' : undefined)};
  border-bottom: 1px solid rgb(169, 169, 169);
  background: ${({ dark = true }) =>
    dark ? '#455a64' : 'rgba(0, 0, 0, 0.05)'};
  padding: 0.5rem;

  & > * {
    white-space: nowrap;
    flex: 1;
    margin: 0.25rem;

    @media (width >= 480px) {
      flex: ${({ separatePrimaryButton }) =>
        separatePrimaryButton ? '0' : undefined};
    }
  }

  & > *:first-child {
    order: ${({ naturalOrder = false }) => (naturalOrder ? undefined : '2')};

    @media (width >= 480px) {
      margin-right: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && naturalOrder ? 'auto' : undefined};
      margin-left: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && !naturalOrder ? 'auto' : undefined};
    }
  }

  & > *:only-child {
    @media (width >= 480px) {
      flex: ${({ centerOnlyChild = true }) => (centerOnlyChild ? 0 : 1)};
      margin: ${({ centerOnlyChild = true }) =>
        centerOnlyChild ? 'auto' : undefined};
      min-width: 33.333%;
    }
  }

  @media print {
    display: none;
  }
`;
