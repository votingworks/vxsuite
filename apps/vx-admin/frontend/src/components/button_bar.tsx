import styled from 'styled-components';

interface Props {
  dark?: boolean;
  primaryRight?: boolean;
  padded?: boolean;
}

export const ButtonBar = styled('nav')<Props>`
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: center;
  justify-content: space-between;
  background: ${({ dark = false }) =>
    dark ? '#333333' : 'rgba(0, 0, 0, 0.05)'};
  padding: ${({ padded = false }) => (padded ? '0.25rem' : undefined)};

  & > * {
    flex-grow: 1;
    margin: 0.25rem;
  }

  & > *:first-child {
    order: ${({ primaryRight = false }) => (primaryRight ? 2 : undefined)};
  }
`;
