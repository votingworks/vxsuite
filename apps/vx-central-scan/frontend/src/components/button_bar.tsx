import styled from 'styled-components';

interface Props {
  dark?: boolean;
  naturalOrder?: boolean;
  secondary?: boolean;
  separatePrimaryButton?: boolean;
  centerOnlyChild?: boolean;
}

export const ButtonBar = styled('nav')<Props>`
  display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  order: ${({ secondary }) => (secondary ? '-1' : undefined)};
  border-bottom: 1px solid rgb(169, 169, 169);
  background: ${({ dark = true }) =>
    dark ? '#455a64' : 'rgba(0, 0, 0, 0.05)'};
  padding: 0.5rem;

  & > *:first-child {
    order: ${({ naturalOrder = false }) => (naturalOrder ? undefined : '2')};
    @media (min-width: 480px) {
      margin-right: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && naturalOrder ? 'auto' : undefined};
      margin-left: ${({ naturalOrder = false, separatePrimaryButton }) =>
        separatePrimaryButton && !naturalOrder ? 'auto' : undefined};
    }
  }

  & > * {
    white-space: nowrap;
    flex: 1;
    margin: 0.25rem;
    @media (min-width: 480px) {
      flex: ${({ separatePrimaryButton }) =>
        separatePrimaryButton ? '0' : undefined};
    }
  }
  & > *:only-child {
    @media (min-width: 480px) {
      flex: ${({ centerOnlyChild = true }) => (centerOnlyChild ? 0 : 1)};
      margin: ${({ centerOnlyChild = true }) =>
        centerOnlyChild ? 'auto' : undefined};
      min-width: 33.333%;
    }
  }
  @media print {
    display: none;
  }
`;
