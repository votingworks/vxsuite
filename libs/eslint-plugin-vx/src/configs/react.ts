import recommended = require('./recommended');

const jsExtensions = ['.js', '.jsx'];
const tsExtensions = ['.ts', '.tsx'];
const allExtensions = jsExtensions.concat(tsExtensions);

export = {
  ...recommended,
  extends: [
    ...recommended.extends
      .map((name) => (name === 'airbnb-base' ? 'airbnb' : name))
      .filter((name) => name !== 'plugin:prettier/recommended'),
    'airbnb/hooks',
    'plugin:react/recommended',
    'plugin:jsx-a11y/recommended',
    'plugin:prettier/recommended', // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
  ],
  settings: {
    ...recommended.settings,
    react: {
      version: 'detect',
    },
  },
  rules: {
    ...recommended.rules,
    'jsx-a11y/label-has-associated-control': [
      'error',
      {
        controlComponents: ['Select'],
      },
    ],
    'react/jsx-filename-extension': ['error', { extensions: allExtensions }],
    'react/jsx-fragments': ['error', 'element'],
    'react/jsx-no-bind': 'off',
    'react/jsx-no-constructed-context-values': 'off',
    'react/jsx-no-useless-fragment': ['error', { allowExpressions: true }],
    'react/jsx-one-expression-per-line': 'off',
    'react/jsx-props-no-spreading': 'off',
    'react/jsx-wrap-multilines': 'off',
    'react/prop-types': 'off',
    'react/require-default-props': 'off',
  },
};
