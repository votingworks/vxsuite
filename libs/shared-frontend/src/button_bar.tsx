import styled from 'styled-components';

export const ButtonBar = styled('nav')`
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid rgb(169, 169, 169);
  background: rgba(0, 0, 0, 0.05);
  padding: 0.25rem;

  & > *:first-child {
    order: 2;
    min-width: 50%;
  }

  & > * {
    flex-grow: 1;
    margin: 0.25rem;
  }
  & > *:only-child {
    @media (min-width: 480px) {
      flex-grow: initial;
      margin: auto;
      min-width: 33.333%;
    }
  }
`;
