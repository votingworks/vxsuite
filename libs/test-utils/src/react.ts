/* eslint-disable no-console -- this file monkey-patches `console` */

/**
 * Suppresses the "Can't perform a React state update on an unmounted component" warning by
 * modifying the `console.error` method to ignore the warning.
 *
 * This is a temporary workaround until we upgrade to React 18.
 */
export function suppressReact17UnmountedWarning(): void {
  const originalConsoleError = console.error;
  console.error = (...args: unknown[]) => {
    if (
      typeof args[0] === 'string' &&
      args[0].startsWith(
        `Warning: Can't perform a React state update on an unmounted component`
      )
    ) {
      return;
    }

    originalConsoleError(...args);
  };
}
