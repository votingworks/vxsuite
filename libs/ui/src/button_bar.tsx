import styled from 'styled-components';

export const ButtonBar = styled('div')`
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: center;
  justify-content: center;
  border-top: ${(p) => p.theme.sizes.bordersRem.hairline}rem solid
    ${(p) => p.theme.colors.foreground};
  padding: ${(p) => p.theme.sizes.minTouchAreaSeparationPx}px;
  gap: ${(p) => p.theme.sizes.minTouchAreaSeparationPx}px;

  & > * {
    flex-grow: 1;
  }

  & > *:first-child {
    order: 2;
    min-width: 40%;
  }

  & > *:only-child {
    @media (min-width: 480px) {
      flex-grow: initial;
      margin: auto;
      min-width: 33.333%;
    }
  }
`;
