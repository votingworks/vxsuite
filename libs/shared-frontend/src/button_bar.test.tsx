import React from 'react';
import { render } from '@testing-library/react';

import { ButtonBar } from './button_bar';

test('renders ButtonBar', () => {
  const { container } = render(<ButtonBar>foo</ButtonBar>);
  expect(container.firstChild).toMatchSnapshot();
});
