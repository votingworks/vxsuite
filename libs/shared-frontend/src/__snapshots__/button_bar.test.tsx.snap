// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ButtonBar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgb(169,169,169);
  background: rgba(0,0,0,0.05);
  padding: 0.25rem;
}

.c0 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  min-width: 50%;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 0.25rem;
}

@media (min-width:480px) {
  .c0 > *:only-child {
    -webkit-box-flex: initial;
    -webkit-flex-grow: initial;
    -ms-flex-positive: initial;
    flex-grow: initial;
    margin: auto;
    min-width: 33.333%;
  }
}

<nav
  class="c0"
>
  foo
</nav>
`;
